tests/static-analysis/assert-alnum.php

```diff
@@ -4,16 +4,36 @@
 
 use Webmozart\Assert\Assert;
 
+/**
+ * @psalm-pure
+ */
+function alnum(string $value): string
+{
+    Assert::alnum($value);
+
+    return $value;
+}
+
+/**
+ * @psalm-pure
+ */
+function nullOrAlnum(?string $value): ?string
```

Comment:
After developing 100+ (lol) type tests after few iterations I came to the following mental framework: the direct tests should only check what the actual function signature can guarantee.

If additionally to that we want to add a test to check some _fundamental pralm properties_ (I will mention it again at least one more time a few comments below) - then it should come to a separate `assert-psalm-xxx` file.

Now back to this very example: 

> We should have a test where a nullOrX leads to @return null at all times

Is it like when you pass a `null` to `Assert::nullOrAlnum` then the `null` should be returned? In that case - it would go to a `assert-psalm-nullStaysNull.php` test, but I'm not sure what value it would add.